### PR TITLE
Add reduce option to huber loss

### PR DESCRIPTION
**Pull request by delta2323**
This PR add `reduce` option to `F.huber_loss`. Rel.  #2558

### CODE REVIEW

File: chainer/functions/loss/huber_loss.py [hunk 1]
```diff
@@ -7,9 +7,15 @@
 
 class HuberLoss(function.Function):
 
-    def __init__(self, delta):
+    def __init__(self, delta, reduce='sum_along_second_axis'):
         self.delta = delta
 
+        if reduce not in ('sum_along_second_axis', 'no'):
+            raise ValueError(
```
Comment (unnonouno) [inline chainer/functions/loss/huber_loss.py]: Add test for this invalid argument.

Comment (delta2323) [inline chainer/functions/loss/huber_loss.py]: I am wondering if there is better option name than `sum_along_second_axis`. Do you have any idea?

Comment (unnonouno) [inline chainer/functions/loss/huber_loss.py]: I think it is a little bit long but is acceptable.

Comment (delta2323) [inline chainer/functions/loss/huber_loss.py]: OK. Then, I left it as it is.

**Pull request merged**